content/docs/reactivesearch/searchbase/overview/apireference.md

```diff
@@ -419,258 +208,91 @@ const searchbase = new Searchbase({
-- **onRequestStatusChange** `(next: string, prev: string) => void`;
-called when request status changes
-- **onSuggestionsRequestStatusChange** `(next: string, prev: string) => void`;
-called when suggestions request status changes
-- **onQueryChange** `(next: string, prev: string) => void`;
-called when query changes
-- **onSuggestionsQueryChange** `(next: string, prev: string) => void`;
-called when suggestions query changes
-- **onMicStatusChange** `(next: string, prev: string) => void`;
-called when mic status changes
-
-##Subscribe to the properties changes
-Although we have events callback which can be used to update the UI based on a particular property changes, the `subscribeToStateChanges` method gives you more control over the UI rendering logic and is more efficient.
-
-### How does it work?
-1. This method is controlled by the `stateChanges` property which can be defined in the [setters](#setters) while updating a particular property. If `stateChanges` is set to `true`, then only the subscribed functions will be called unlike events callback which gets called every time when a property changes it's value.<br/>
-So basically, `subscribeToStateChanges` provides more control over the events callback in a way that you can define whether to update the UI or not while setting a particular property's value.<br/>
-2. You can define the properties for which you want to update the UI.
-3. It allows you to register multiple callback functions for search state updates.
-
-### Usage
+
+Additionally, you can override the global configurations by defining it for a particular component. For example, to register a component with a different `index` name.
+
 ```ts
-subscribeToStateChanges(
-    fn: Function,
-    propertiesToSubscribe?: string | Array<string>
-): void
+const searchBase = new SearchBase({
+	index: 'gitxplore-app',
+	url: 'https://@arc-cluster-appbase-demo-6pjy6z.searchbase.io',
+	credentials: 'a03a1cb71321:75b6603d-9456-4a5a-af6b-a487b309eb61',
+});
+
+searchBase.register('search-component', {
+    index: 'gitxplore-app-2',
+	dataField: ['title', 'description'],
+	value: '',
+});
 ```
-You can use the `subscribeToStateChanges` method of `Searchbase` class to subscribe to the state changes of the properties. <br/>
-A common use-case is to subscribe a component or DOM element to a particular property or a set of properties & update the UI according to the changes. <br/>
-The callback function accepts an object in the following shape:
-```js
-{
-    [propertyName]: { // property name for example, `results`
-        prev: any; // previous value of the property
-        next: any; // next value of the property
-    }
-}
+
+#### unregister
+
+```ts
+unregister(id: string): void
 ```
 
-Let's check this example to bind a React component with the `results` property change.
+This method is useful to un-register a component by `id`. It is a good practice to un-register(remove) an un-mounted/un-used component to avoid the side-effects.
```

Comment:
```suggestion
This method is useful to unregister a component by `id`. It is a good practice to unregister (remove) an unmounted/unused component to avoid any side-effects.
```